spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/abstractnn/AbstractModule.scala

```diff
@@ -580,5 +602,23 @@ abstract class AbstractModule[A <: Activity: ClassTag, B <: Activity: ClassTag,
               ): Array[(ValidationResult, ValidationMethod[T])] = {
     Validator(this, dataSet).test(vMethods)
   }
+
+  private var nameToSubModules: Map[String, Module[T]] = _
+  /**
+   * get all submodules map, use the module name as key
+   */
+  def getSubModules(): Map[String, Module[T]] = {
+    if (nameToSubModules == null) nameToSubModules = Utils.getNamedModules(this)
```

Comment:
this makes no sense at all. Why a layer should have a method named getSubModules? A layer don't have sub modules.